sql/sem/builtins: TestSerialNormalizationWithUniqueUnorderedID failed

sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10862560?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10862560?buildTab=artifacts#/) on release-23.1 @ [a1062b43480d456afec57e6df469c87f9f9b0162](https://github.com/cockroachdb/cockroach/commits/a1062b43480d456afec57e6df469c87f9f9b0162):


```
I230711 12:02:26.912038 2278379 13@kv/kvserver/replicate_queue.go:847  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 385  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230711 12:02:26.948513 1778082 3@pebble/event.go:717  [n1,s1,pebble] 386  [JOB 9] WAL created 000008
I230711 12:02:27.909917 2363948 13@kv/kvserver/replicate_queue.go:847  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 387  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    builtins_test.go:166: Key counts in each split range
    builtins_test.go:168: range 0: 1321
    builtins_test.go:168: range 1: 1320
    builtins_test.go:168: range 2: 1319
    builtins_test.go:168: range 3: 1337
    builtins_test.go:168: range 4: 1198
    builtins_test.go:168: range 5: 1158
    builtins_test.go:168: range 6: 1180
    builtins_test.go:168: range 7: 1167
I230711 12:02:28.455021 1791849 sql/stats/automatic_stats.go:572  [T1,n1] 388  quiescing auto stats refresher
W230711 12:02:28.454988 1788665 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 389  exiting heartbeat loop
I230711 12:02:28.455807 1792253 sql/stats/automatic_stats.go:612  [T1,n1] 390  quiescing stats garbage collector
W230711 12:02:28.456471 2382230 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 391  failed to gc transaction record: could not GC completed transaction anchored at /Table/106/1/5272932214584901633/0: node unavailable; try another peer
W230711 12:02:28.456457 1788665 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 393  exiting heartbeat loop with error: node unavailable; try another peer
I230711 12:02:28.456149 1774881 server/start_listen.go:103  [T1,n1] 392  server shutting down: instructing cmux to stop accepting
I230711 12:02:28.455978 1876297 rpc/context.go:2318  [T1,n1,rnode=1,raddr=127.0.0.1:41001,class=default,rpc] 394  connection heartbeat loop ended with err: <nil>
E230711 12:02:28.457297 1788665 server/server_sql.go:509  [T1,n1] 395  failed to run update of instance with new session ID: node unavailable; try another peer
I230711 12:02:28.457497 1827841 jobs/registry.go:1588  [T1,n1] 396  AUTO SPAN CONFIG RECONCILIATION job 881480630170320897: stepping through state succeeded
E230711 12:02:28.457813 1827841 jobs/registry.go:989  [T1,n1] 397  error getting live session: node unavailable; try another peer
W230711 12:02:28.469031 2136384 jobs/adopt.go:518  [T1,n1] 398  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: ed2b2a51], [can-forward-ts] RPC error: node unavailable; try another peer
I230711 12:02:28.474442 1117166 server/server_controller_orchestration.go:263  [T1,n1] 399  server controller shutting down ungracefully
I230711 12:02:28.474507 1117166 server/server_controller_orchestration.go:274  [T1,n1] 400  waiting for tenant servers to report stopped
W230711 12:02:28.475283 1117166 server/server_sql.go:1704  [T1,n1] 401  server shutdown without a prior graceful drain
    builtins_test.go:190: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins_test.go:190
        	Error:      	Received unexpected error:
        	            	chiSquared value of 36.182400 must be less than criticalVal 35.258500 to guarantee distribution is relatively uniform
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins_test.go:184
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:195
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.Options.Do
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:172
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:194
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins_test.go:110
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) chiSquared value of 36.182400 must be less than criticalVal 35.258500 to guarantee distribution is relatively uniform
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestSerialNormalizationWithUniqueUnorderedID
--- FAIL: TestSerialNormalizationWithUniqueUnorderedID (17.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSerialNormalizationWithUniqueUnorderedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29628

--- a/pkg/sql/sem/builtins/builtins_test.go
+++ b/pkg/sql/sem/builtins/builtins_test.go
@@ -124,7 +124,7 @@ CREATE TABLE t (
   j INT
 )`)
 
-		numberOfRows := 10000
+		numberOfRows := 100000
 
 		// Insert rows.
 		tdb.Exec(t, fmt.Sprintf(`

